jwriter/writer.go

```diff
@@ -83,6 +83,23 @@ func (w *Writer) Raw(data []byte, err error) {
 	}
 }
 
+// RawText encloses raw binary data in quotes and appends in to the buffer.
+// Useful for calling with results of MarshalText-like functions.
+func (w *Writer) RawText(data []byte, err error) {
```

Comment:
You can't just do that, you need to use String() to properly handle escapes.

If you do care about extra allocation here, you can use unsafe []byte -> string conversion, but it doesn't make much sense since TextMarshaler is very likely to do an allocation here anyway. In fact, net.IP's MarshalText function should do quite a lot of them due to using string '+' operator.